rust/lance/src/index.rs

```diff
@@ -3983,4 +3990,545 @@ mod tests {
             .to_string()
             .contains("not found in source dataset"));
     }
+
+    #[tokio::test]
+    async fn test_vector_index_on_nested_field_with_dots() {
+        let dimensions = 16;
+        let num_rows = 256;
+
+        // Create schema with nested field containing dots in the name
+        let struct_field = Field::new(
+            "embedding_data",
+            DataType::Struct(
```

Comment:
currently there is no nested field datagen util. I will put another PR to refactor these tests.